sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/NetworkInterface.java

```diff
@@ -591,6 +694,301 @@ interface Update
             UpdateStages.WithIPConfiguration,
             UpdateStages.WithLoadBalancer,
             UpdateStages.WithAcceleratedNetworking,
-            UpdateStages.WithApplicationSecurityGroup {
+            UpdateStages.WithApplicationSecurityGroup,
+            UpdateStages.WithPublicIPAddressDeleteOptions {
+    }
+
+    /** An interface representing a model's ability to reference a public IP address. */
+    interface WithDefinitionSecondaryOrUpdatePublicIPAddress {
```

Comment:
This interface seems to have introduced more code/duplication?

Reply:
The existence of duplicate codes is inevitable because they belong to `DefinitionStages/UpdateStages/UpdateDefinitionStages` respectively.

Reply:
Emmm, I see. Seems worse than before to me.

Reply:
It is not as good as the version repaired on [3/15](https://github.com/Azure/azure-sdk-for-java/pull/39096/commits/c45ef75ebc91613257d588aef01c010b8cbb3607).